compose/compose.go

```diff
@@ -572,6 +575,85 @@ func fatal(err error) {
 	os.Exit(1)
 }
 
+func makeDir(path string) error {
+	_, err := os.Stat(path)
+	if os.IsNotExist(err) {
+		if errs := os.MkdirAll(path, 0755); errs != nil {
+			fatal(errors.Errorf("Couldn't create directory %v. Error: %v", path, errs))
+		}
+	} else if err != nil {
+		fatal(errors.Errorf("Something went wrong while checking if directory %v still exists. Error: %v",
+			path, err))
+	}
+	return nil
+}
+
+func copyFile(src, dst string) error {
+	var err error
+	var srcfd *os.File
+	var dstfd *os.File
+	var srcinfo os.FileInfo
+
+	if srcfd, err = os.Open(src); err != nil {
+		fatal(errors.Errorf("Error in opening source file %v. Error: %v", src, err))
+		return err
+	}
+	defer srcfd.Close()
+
+	if dstfd, err = os.Create(dst); err != nil {
+		fatal(errors.Errorf("Error in creating destination file %v. Error: %v", dst, err))
+		return err
+	}
+	defer dstfd.Close()
+
+	if _, err = io.Copy(dstfd, srcfd); err != nil {
+		fatal(errors.Errorf("Error in copying source file %v to destination file %v. Error: %v",
+			src, dst, err))
+		return err
+	}
+	if srcinfo, err = os.Stat(src); err != nil {
+		fatal(errors.Errorf("Error in doing stat of source file %v. Error: %v", src, err))
+		return err
+	}
+	return os.Chmod(dst, srcinfo.Mode())
+}
+
+func copyDir(src string, dst string) error {
+	var err error
+	var fds []os.FileInfo
+	var srcinfo os.FileInfo
+
+	if srcinfo, err = os.Stat(src); err != nil {
```

Comment:
Nitpick: when naming variables, functions, etc., it's best to stick to what go calls [mixedCaps](https://golang.org/doc/effective_go#mixed-caps) to stay idiomatic-ish.

Reply:
I recommend looking at the go std libraries for a good reference on golang style, that what I typically refer to. For ex: [strings](https://github.com/golang/go/blob/master/src/strings/strings.go)

Reply:
Also for the formatting issues you mentioned in this PR, check out `go fmt`

Reply:
Moved variable to mixed caps. Yes, later to fix formatting I used `gofmt -s compose.go`.